addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -337,7 +337,6 @@ export class LinkPlugin extends Plugin {
                     after = linkElement.nextSibling;
                 }
                 this.shared.setCursorEnd(linkElement);
-                this.dispatch("ADD_STEP");
```

Comment:
We need this step because if we select part of a link + some text, click the link button, we won't be able to undo this step